### PR TITLE
Add health checks

### PR DESCRIPTION
**Pull request by Kerram**

### CODE REVIEW

File: health_check.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            lock.acquire()
+            hearthbeat(linter)
+        finally:
+            lock.release()
+
+
+def hearthbeat(linter):
+    global failures
+
+    response = requests.get(f"{linter.address}/", timeout=timeout).json()
```
Comment (mkmkl93) [inline health_check.py:60]: Do we really need different timeout configurations for tests and production environment?

**Review by mkmkl93 — CHANGES_REQUESTED** [review]

File: e2e_tests.py [hunk 2]
```diff
@@ -324,6 +338,62 @@ def test_opening_application_on_already_occupied_socket(self):
         deploy_utils.stop_fast_api_app(machine_manager_process)
         deploy_utils.stop_fast_api_app(load_balancer_process)
 
+    def test_health_check_when_linters_work(self):
+        try:
+            self.create_linter_instances(2, E2eTests.v1)
+            self.start_health_check()
+            time.sleep(2)
+
+            time.sleep(E2eTests.health_check_timeout * 2 + E2eTests.health_check_repetition_period)
```
[the remark this answers is not in the document]
Comment (Kerram) [reply e2e_tests.py]: Done.

File: health_check.py (hunk 1 above)
Comment (Kerram) [reply health_check.py:60]: Yes, I want tests to run fast enough and to control the timeout there.

File: health_check_api.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import requests
+
+
+def get_failures(url) -> int:
```
[the remark this answers is not in the document]
Comment (Kerram) [reply health_check_api.py:4]: Yes.

File: test/test_e2e.py [hunk 4]
```diff
@@ -42,6 +42,7 @@ def setUp(self) -> None:
         os.environ["MACHINE_MANAGER_DEPLOY_BACKEND"] = "local"
         E2eTests.set_linter_test_aps_supported_versions()
         self.machine_manager_process, self.machine_manager_url = deploy_utils.start_fast_api_app("machine_manager")
+        print(self.machine_manager_url)
```
Comment (mkmkl93) [inline test/test_e2e.py:45]: I think that we shouldn't use print

File: test/test_health_check.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    def stop_health_check(self):
+        deploy_utils.stop_fast_api_app(self.hc_process)
+        os.environ.pop("HEALTH_CHECK_TIMEOUT")
+        os.environ.pop("HEALTH_CHECK_REPETITION_PERIOD")
+
+    def wait_for_hearthbeat(self):
+        # It is not ideal, but it works. We want 2 times timeout because we have 2 machines in tests and one repetition
+        # period to ensure hearthbeat will be called at least once.
+        time.sleep(HealthCheckTests.health_check_timeout * 2 + HealthCheckTests.health_check_repetition_period)
```
Comment (mkmkl93) [inline test/test_health_check.py:36]: Thanks for the comment

**Review by mkmkl93 — CHANGES_REQUESTED** [review]

File: test/test_e2e.py (hunk 4 above)
Comment (Kerram) [reply test/test_e2e.py:45]: Right, thx!

**Review by mkmkl93 — APPROVED** [review]

**Pull request merged**